zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -190,10 +210,30 @@ void closeAndRemoveInterpreterGroup(String noteId) {
     }
   }
 
+  void closeAndRemoveInterpreterGroupByUser(String user) {
+    if (user.equals("anonymous")) {
```

Comment:
it'd probably be good to have a const for "anonymous" and "" - easier to avoid typos etc.